docs/src/model_setup/boundary_conditions.md

```diff
@@ -488,3 +488,54 @@ Oceananigans.FieldBoundaryConditions, with boundary conditions
 └── immersed: ImmersedBoundaryCondition with west=Nothing, east=Nothing, south=Nothing, north=Nothing, bottom=Value, top=Nothing
 ```
 
+A boundary condition that depends on the fields may be prescribed using the `immersed` keyword argument in [`FieldBoundaryConditions`](@ref). Let's see here how we can add linear bottom drag on `u` in the case we have
+an immersed boundary. In essence, we'd need to boundary conditions: one that would apply at the grid's bottom,
+and when that would apply on the immersed boundary.
+
+First let's create the boundary condition for the grid's bottom:
+
+```jldoctest immersed_bc
+julia> @inline linear_drag(x, y, t, u) = - 0.2 * u
+linear_drag (generic function with 1 method)
+
+julia> drag_u = FluxBoundaryCondition(linear_drag, field_dependencies=:u)
+FluxBoundaryCondition: ContinuousBoundaryFunction linear_drag at (Nothing, Nothing, Nothing)
+```
+
+Then let's create the immersed boundary condition:
+
+```jldoctest immersed_bc
+julia> @inline immersed_linear_drag(x, y, z, t, u) = - 0.2 * u
+immersed_linear_drag (generic function with 1 method)
+
+julia> immersed_drag_u = FluxBoundaryCondition(immersed_linear_drag, field_dependencies=:u)
```

Comment:
```suggestion
julia> @inline linear_drag(x, y, z, t, u) = - 0.2 * u
linear_drag (generic function with 1 method)

julia> immersed_drag_u = FluxBoundaryCondition(linear_drag, field_dependencies=:u)
```

Just a suggestion, but using the same name `linear_drag` for both functions makes the code cleaner and illustrates that we can rely on multiple dispatch to use the correct method

Reply:
One downside is that using the same function name might be confusing for people who aren't well-versed in multiple dispatch

Reply:
I left it w different names.